### PR TITLE
Assorted GE debugger crash fixes

### PR DESCRIPTION
**Pull request by hrydgard**
Also fixes issue where the rewind machinery would run even during pause.

Thanks Nemoumbra for instructions.

Fixes most of, or maybe all of #18569

### CODE REVIEW

File: GPU/GLES/GPU_GLES.cpp [hunk 1]
```diff
@@ -248,7 +248,7 @@ void GPU_GLES::BeginHostFrame() {
 	textureCache_->StartFrame();
 
 	// Save the cache from time to time. TODO: How often? We save on exit, so shouldn't need to do this all that often.
-	if (shaderCachePath_.Valid() && (gpuStats.numFlips & 4095) == 0) {
+	if (shaderCachePath_.Valid() && (gpuStats.numFlips & 32767) == 0 && coreState == CORE_RUNNING) {
```
Comment (JustAWhiteBloodCell) [inline GPU/GLES/GPU_GLES.cpp:251]: Might be worth documenting this constant.

File: Windows/GEDebugger/CtrlDisplayListView.cpp [hunk 2]
```diff
@@ -306,6 +310,10 @@ void CtrlDisplayListView::onMouseDown(WPARAM wParam, LPARAM lParam, int button)
 
 void CtrlDisplayListView::onMouseUp(WPARAM wParam, LPARAM lParam, int button)
 {
+	if (!validDisplayList) {
```
Comment (JustAWhiteBloodCell) [inline Windows/GEDebugger/CtrlDisplayListView.cpp:313]: nit: Inconsistent bracing style, relative to nearby code

Comment (Nemoumbra) [inline Windows/GEDebugger/CtrlDisplayListView.cpp:313]: Do we need both this and my state checks a few lines down there?

**Review by JustAWhiteBloodCell — APPROVED** [review]

File: Windows/GEDebugger/CtrlDisplayListView.cpp (hunk 2 above)
Comment (hrydgard) [reply Windows/GEDebugger/CtrlDisplayListView.cpp:313]: Apparently we do, currently, but I'm changing it to check gpuDebug too. Now not needed.

Comment (hrydgard) [reply Windows/GEDebugger/CtrlDisplayListView.cpp:313]: it's a mix of bracing styles in this file, and we're moving towards this one. I'm keeping it.

Comment (JustAWhiteBloodCell) [reply Windows/GEDebugger/CtrlDisplayListView.cpp]: Makes sense!

**Pull request merged**